runtime/common/src/remote_government/mod.rs

```diff
@@ -0,0 +1,211 @@
+	pub type PreviousBridgeFinalized<T> = StorageValue<_, BlockNumberFor<T>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn emergency)]
+	pub type Emergency<T> = StorageValue<_, bool, ValueQuery, DefaultForEmergency>;
+	#[pallet::type_value]
+	pub fn DefaultForEmergency() -> bool {
+		false
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(now: BlockNumberFor<T>) -> Weight {
+			// If emergency, we check the sync state each block.
+			if Self::emergency() {
+				if Self::check_sync() {
+					<Emergency<T>>::kill();
+
+					Self::deposit_event(Event::Recovery);
+
+					T::DbWeight::get().reads_writes(2, 2)
+				} else {
+					T::DbWeight::get().reads(2)
+				}
+			} else if (now % T::CheckInterval::get()).is_zero() {
+				if Self::check_sync() {
+					T::DbWeight::get().reads_writes(2, 1)
+				} else {
+					<Emergency<T>>::put(true);
+
+					Self::deposit_event(Event::Emergency);
+
+					T::DbWeight::get().reads_writes(2, 2)
+				}
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+	}
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
+			(dispatch_info.weight.saturating_add(10_000), dispatch_info.class)
+        })]
+		#[transactional]
+		pub fn emergency_safeguard(
+			origin: OriginFor<T>,
+			call: AnyCall<T>,
+		) -> DispatchResultWithPostInfo {
+			if !Self::emergency() {
+				Err(<Error<T>>::EmergencyOnly)?;
+			}
+
+			T::EmergencySafeguardOrigin::ensure_origin(origin)?;
+
+			let res = call.dispatch_bypass_filter(RawOrigin::Root.into());
+
+			Self::deposit_event(Event::EmergencySafeguardDone {
+				result: res.map(|_| ()).map_err(|e| e.error),
+			});
+
+			// Sudo user does not pay a fee.
+			Ok(Pays::No.into())
+		}
+
+		/// Handle relay message sent from the source backing pallet with relay message
+		#[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
+			(dispatch_info.weight.saturating_add(10_000), dispatch_info.class)
+        })]
+		#[transactional]
+		pub fn enact_remote_call(
+			origin: OriginFor<T>,
+			call: AnyCall<T>,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			let source_root = Self::derived_source_root();
+
+			ensure!(source_root == sender, Error::<T>::RequireSourceRoot);
+
+			let res = call.dispatch_bypass_filter(RawOrigin::Root.into());
+
+			Self::deposit_event(Event::RemoteCallEnacted {
+				result: res.map(|_| ()).map_err(|e| e.error),
+			});
+
+			// Sudo user does not pay a fee.
+			Ok(Pays::No.into())
+		}
+	}
+	impl<T: Config> Pallet<T> {
+		fn check_sync() -> bool {
+			<PreviousBridgeFinalized<T>>::try_mutate(|previous_bridge_finalized| {
+				let best_finalized = T::BridgeFinalized::get();
+
+				if *previous_bridge_finalized == best_finalized {
+					Err(())
```

Comment:
Why return error here?

Reply:
If they are the same, skip storage writing.